priority_que.py

```diff
@@ -0,0 +1,41 @@
+import bisect
+
+from binheap import BinaryHeap
+
+
+class PriorityQue:
+    def __init__(self, values=None):
+        self._storage = BinaryHeap("max")
+        if values:
+            for priority, value in values:
+                self._storage.push(Node(priority, value))
+
+    def __len__(self):
+        return self._storage._size
+
+    def insert(self, value, priority=0):
+        self._storage.push(Node(priority, value))
+
+    def pop(self):
+        try:
+            return self._storage.pop().value
+        except IndexError:
```

Comment:
We also missed this in the BH review. It shouldn't raise an IndexError, and neither should the Queue.  Neither of them have the concept of an Index.

Reply:
Why are the tests passing even though you are catching an IndexError here?